Documentation/gettingstarted/kind.rst

```diff
@@ -0,0 +1,224 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _gs_kind:
+
+**************************
+Getting Started Using Kind
+**************************
+
+This guide uses `kind <https://kind.sigs.k8s.io/>`_ to demonstrate deployment
+and operation of Cilium in a multi-node Kubernetes cluster.
+
+Kind requires docker to be installed and running.
+
+Install Docker, kubectl, & kind
+===============================
+
+1. Install ``docker`` stable as described in: `Install Docker Engine <https://docs.docker.com/engine/install/>`_
+
+2. Install ``kubectl`` version >= v1.14.0 as described in the `Kubernetes Docs <https://kubernetes.io/docs/tasks/tools/install-kubectl/>`_.
+
+3. Install ``kind`` >= v0.7.0 per kind documentation: `Installation and Usage <https://kind.sigs.k8s.io/#installation-and-usage>`_
+
+Kind Configuration
+==================
+
+Kind doesn't use flags for configuration. Instead it uses YAML configuration that is very similar to Kubernetes.
+
+Create a ``kind-config.yaml`` file based on the following template. The template will create 3 node + 1 apiserver
+cluster with the latest version of kubernetes from when the kind release was created.
+
+.. code:: yaml
+
+    kind: Cluster
+    apiVersion: kind.x-k8s.io/v1alpha4
+    nodes:
+    - role: control-plane
+    - role: worker
+    - role: worker
+    - role: worker
+    networking:
+      disableDefaultCNI: true
+
+To change the version of kubernetes being run,  ``image`` has to be defined for each node. See
+the `Node Configration <https://kind.sigs.k8s.io/docs/user/configuration/#nodes>`_ documentation.
+
+Start Kind
+==========
+
+Pass the ``kind-config.yaml`` you created with the ``--config`` flag of kind.
+
+.. code:: bash
+
+    kind create cluster --config=kind-config.yaml
+
+This will add a ``kind-kind`` context to ``KUBECONFIG`` or if unset ``${HOME}/.kube/config``
+
+.. code:: bash
+
+    kubectl cluster-info --context kind-kind
+
+.. _kind_install_cilium:
+
+Install Cilium
+==============
+
+.. include:: k8s-install-download-release.rst
+
+.. note::
+
+    As Kind is running nodes as containers in Docker, they're sharing your host machines' kernel.
+    So we have to disable :ref:`host-services` otherwise kind clusters will clobber each other
+    and/or Cilium won't be able to restart until the host is restarted.
+
+Install Cilium release via Helm:
+
+.. parsed-literal::
+
+   helm install cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --set global.nodeinit.enabled=true \\
+      --set global.kubeProxyReplacement=partial \\
+      --set global.externalIPs.enabled=true \\
+      --set global.nodePort.enabled=true \\
+      --set global.hostPort.enabled=true
```

Comment:
Oh, by the way, which of these did you find were necessary and which of them are more related to your setup?

I'd expect that anyone could be successful with `kind` perhaps with as little as the `nodeinit` parameter at least for the single-cluster multi-node case.

Reply:
Reading the documentation again I guess this is where your hostservices bit comes in, the `kubeProxyReplacement=partial` disables various flags by default then you're manually re-enabling some of them which are known to work.

I wonder if we could simplify this further.

Reply:
Yup, exactly. I'm trying to keep feature parity of `probe` while only disabling `hostServices`.